modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache

```diff
@@ -101,6 +101,13 @@ export class BaseAPI {
     }
 };
 
+export class ResponseError extends Error {
```

Comment:
note that this comes with some gotchas and generally won't work if your TS compile target is `es5`.

e.g. see https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work

Reply:
As a workaround, you can match by error name if your code needs to be compiled to es5

```ts
if (err.name === 'ResponseError') {
   ...
}
```